jib-core/src/main/java/com/google/cloud/tools/jib/registry/ManifestPusher.java

```diff
@@ -53,6 +58,62 @@ public BlobHttpContent getContent() {
     return Collections.emptyList();
   }
 
+  @Override
+  public Void handleHttpResponseException(HttpResponseException httpResponseException)
+      throws HttpResponseException, RegistryErrorException {
+    // docker registry 2.0 and 2.1 returns:
+    //   400 Bad Request
+    //   {"errors":[{"code":"TAG_INVALID","message":"manifest tag did not match URI"}]}
+    // docker registry:2.2 returns:
+    //   400 Bad Request
+    //   {"errors":[{"code":"MANIFEST_INVALID","message":"manifest invalid","detail":{}}]}
+    // quay.io returns:
+    //   415 UNSUPPORTED MEDIA TYPE
+    //   {"errors":[{"code":"MANIFEST_INVALID","detail":
+    //   {"message":"manifest schema version not supported"},"message":"manifest invalid"}]}
+
+    if (httpResponseException.getStatusCode() != HttpStatus.SC_BAD_REQUEST
+        && httpResponseException.getStatusCode() != HttpStatus.SC_UNSUPPORTED_MEDIA_TYPE) {
+      throw httpResponseException;
+    }
+
+    // TODO turn deserializing the error-code into a library method
+
+    // Obtain the error response code.
+    String errorContent = httpResponseException.getContent();
+    if (errorContent == null) {
+      throw httpResponseException;
+    }
+
+    try {
+      ErrorResponseTemplate errorResponse =
+          JsonTemplateMapper.readJson(errorContent, ErrorResponseTemplate.class);
+      List<ErrorEntryTemplate> errors = errorResponse.getErrors();
+      if (errors.size() == 1) {
+        String errorCodeString = errors.get(0).getCode();
+        if (errorCodeString == null) {
+          // Did not get an error code back.
+          throw httpResponseException;
+        }
+        ErrorCodes errorCode = ErrorCodes.valueOf(errorCodeString);
```

Comment:
I noticed `valueOf(String)` will throw `IllegalArgumentException` if there is no matching enum, so I think we should catch that.

Reply:
Good catch.  Will handle this in #937 as this is a lightly modified block taken from elsewhere.